pkg/ovsdb/ovsdb.go

```diff
@@ -221,6 +222,53 @@ func (self *OvsDriver) GetOvsPortForContIface(contIface, contNetnsPath string) (
 	return fmt.Sprintf("%v", port["name"]), true, nil
 }
 
+func (self *OvsDriver) FindInterfacesWithError() ([]string, error) {
+	selectOp := libovsdb.Operation{
+		Op:      "select",
+		Columns: []string{"name", "error"},
+		Table:   "Interface",
+	}
+	transactionResult, err := self.ovsdbTransact([]libovsdb.Operation{selectOp})
+	if err != nil {
+		return nil, err
+	}
+	if len(transactionResult) != 1 {
+		return nil, fmt.Errorf("no transaction result")
+	}
+	operationResult := transactionResult[0]
+	if operationResult.Error != "" {
+		return nil, fmt.Errorf(operationResult.Error)
+	}
+
+	var names []string
+	for _, row := range operationResult.Rows {
+		if isEmpty(row["error"]) {
+			continue
+		}
+		log.Printf("iface %v contains error: %v", row["name"], row["error"])
+		names = append(names, fmt.Sprintf("%v", row["name"]))
+	}
+	return names, nil
+}
+
+func isEmpty(v interface{}) bool {
+	switch x := v.(type) {
+	case []interface{}:
+		if len(x) == 0 {
+			return true
+		}
+		if x[0] == "set" {
+			y, ok := x[1].([]interface{})
+			return !ok || len(y) == 0
+		}
+		return true
+	case string:
+		return len(x) == 0
+	default:
```

Comment:
Frankly, I don't know. So as a safety measure I added the default. Returning `true` would signal that the row contains a value. Since I use it for the `error` row this would mean the row contains an error.

I think it is better to rename the function to `hasError` and pass in the whole `row`. The function itself then checks the `error` key. This way the code is more obvious.

Reply:
That sounds clean! Awesome :)

Reply:
Changed :slightly_smiling_face: